conmon/conmon.c

```diff
@@ -1337,6 +1343,10 @@ int main(int argc, char *argv[])
 	add_argv(runtime_argv, opt_cid, NULL);
 	end_argv(runtime_argv);
 
+	signal(SIGTERM, on_sig_exit);
+	signal(SIGQUIT, on_sig_exit);
+	signal(SIGINT, on_sig_exit);
```

Comment:
I like installing these before the `fork` call (like you have here), but I don't think we want them inside the child process post-fork, because it means a `TERM`, etc. sent to the container process would be ignored.  I'd rather have:

* Set the handlers (like you have here)
* Block the signals with `sigprocmask`.
* `fork`
    * In the child:
        * Set `PDEATHSIG` to `KILL`.  This will be cleared if/when the runtime `fork`s, but it will survive the initial `execv`, so setting it still gives us some useful protection against `conmon` dying.
        * Uninstall the signal handlers.
        * Unblock the signals.
    * In the parent:
        * Unblock the signals.

Alternatively, you could:

* Block the signals with `sigprocmask`.
* `fork`
    * In the child:
        * Set `PDEATHSIG` to `KILL`.
        * Unblock the signals.
    * In the parent:
        * Set the handlers.
        * Unblock the signals.

Reply:
good idea.  I've amended it in a new version